src/rust/lib/graph-editor/src/component/node/port/output.rs

```diff
@@ -0,0 +1,682 @@
+            let BaseShapeData{ port_area,hover_area, .. } = base_shape_data;
+
+            // FIXME: Use colour from style and apply transparency there.
+            let color     = Var::<color::Rgba>::from("srgba(0.25,0.58,0.91,input_opacity)");
+            let port_area = port_area.fill(color);
+
+            (port_area + hover_area).into()
+        }
+    }
+
+    impl PortShapeApi for Shape {
+        fn set_grow(&self, grow_value:f32) {
+            self.grow.set(grow_value)
+        }
+
+        fn set_opacity(&self, opacity:f32) {
+            self.opacity.set(opacity)
+        }
+    }
+}
+
+/// Helper enum that handles the distinction between a single shape output area and a multi port
+/// output area.
+#[derive(Clone,Debug)]
+enum ShapeView {
+    Single { view  : component::ShapeView<multi_port_area::Shape>      },
+    Multi  { views : Vec<component::ShapeView<multi_port_area::Shape>> },
+}
+
+impl ShapeView {
+    /// Constructor.
+    fn new(number_of_ports:NonZeroU32, logger:&Logger, scene:&Scene) -> Self {
+        let number_of_ports = number_of_ports.get();
+        if number_of_ports == 1 {
+            ShapeView::Single { view: component::ShapeView::new(&logger,&scene) }
+        } else {
+            let mut views = Vec::default();
+            let number_of_ports = number_of_ports as usize;
+            views.resize_with(number_of_ports,|| component::ShapeView::new(&logger,&scene));
+            ShapeView::Multi { views }
+        }
+    }
+
+    /// Set up the frp for all ports.
+    fn init_frp(&self, port_frp:PortFrp) {
+        match self {
+            ShapeView::Single {view}  => init_port_frp(&view, PortId::new(0),port_frp),
+            ShapeView::Multi  {views} => {
+                views.iter().enumerate().for_each(|(index,view)| {
+                    init_port_frp(&view, PortId::new(index),port_frp.clone_ref())
+                })
+            }
+        }
+    }
+
+    /// Resize all the port output shapes to fit the new layout requirements for thr given
+    /// parameters.
+    fn update_shape_layout_based_on_size_and_gap(&self, size:Vector2<f32>, gap_width:f32) {
+        match self {
+            ShapeView::Single { view }   => {
+                let shape = &view.shape;
+                shape.sprite.size.set(size);
+            }
+            ShapeView::Multi { views }   => {
+                let port_num  = views.len() as f32;
+                for (index, view) in views.iter().enumerate(){
+                    let shape = &view.shape;
+                    shape.sprite.size.set(size);
+                    shape.index.set(index as f32);
+                    shape.port_num.set(port_num);
+                    shape.padding_left.set(gap_width * 0.5);
+                    shape.padding_right.set(-gap_width * 0.5);
+                }
+                views[0]              .shape.padding_left .set(0.0);
+                views[views.len() - 1].shape.padding_right.set(0.0);
```

Comment:
after we change the shapes to be drawn always with (0,0) in the center of the shapes, these should not be needed anymore (there's should be other simplifications as well).